pkg/webhook/mutation/pod/metadata/mutator.go

```diff
@@ -36,7 +38,19 @@ func (mut *Mutator) Enabled(request *dtwebhook.BaseRequest) bool {
 		request.DynaKube.FeatureAutomaticInjection())
 	enabledOnDynakube := request.DynaKube.MetaDataEnrichmentEnabled()
 
-	return enabledOnPod && enabledOnDynakube
+	matchesNamespace := true // if no namespace selector is configured, we just pass set this to true
+
+	if request.DynaKube.MetadataEnrichmentNamespaceSelector().Size() > 0 {
+		selector, err := metav1.LabelSelectorAsSelector(request.DynaKube.MetadataEnrichmentNamespaceSelector())
+
+		if err != nil {
+			matchesNamespace = false //nolint:ineffassign,wastedassign
```

Comment:
since nothing else is done in the error case, the assignment is "wasted" as `matchesNamespace` gets a value after the `if` anyway

Reply:
adressed in https://github.com/Dynatrace/dynatrace-operator/pull/3199/commits/a4cfb344db2357b76fd158e5284e7454fef9ce10